retrieval/target.go

```diff
@@ -549,6 +563,38 @@ func (t *Target) scrape(appender storage.SampleAppender) error {
 	return err
 }
 
+func (t *Target) report(app storage.SampleAppender, start time.Time, duration time.Duration, err error) {
+	t.status.setLastScrape(start)
```

Comment:
Do you want to remove the other calls to `setLastScrape()` then? The ones in  https://github.com/prometheus/prometheus/blob/463dd3ea06865e3e3c671f4e5ef7a97ad5f82ddd/retrieval/target.go#L436 and https://github.com/prometheus/prometheus/blob/463dd3ea06865e3e3c671f4e5ef7a97ad5f82ddd/retrieval/target.go#L456. But this then also changes the semantics of the "last scrape" slightly, since it'll no longer recorded when skipping during throttling. Maybe intended?